common/scala/src/main/scala/whisk/core/entity/WhiskPackage.scala

```diff
@@ -140,6 +148,20 @@ object WhiskPackage
     val bindingFieldName = "binding"
     override val collectionName = "packages"
 
+    /**
+     * an utility function that traverses all potential binding references and returns the resolved package
+     * @param pkg the package for which the resolution to be applied
+     * @return the same package if there is no binding, or the fully resolved package
+     */
+    def resolveBinding(entityStore: EntityStore, pkg: DocId)(
```

Comment:
Isn't there a layer on top of this which given a sequence of strings resolves those into fully qualified names?